### DISCUSSION TITLE
helm pull timouts in argocd, works fine on host.

### DISCUSSION THREAD

**Discussion by VladoPortos** [argoproj/argo-cd#15842] [Q&A] [answered]
Hi all,

I'm getting timeouts with helm in argocd. I can't really explain it why, since helm on the host is working super fast. Its a single node k3s cluster installed today. So everything is the latest version. 

Example of error
```
time="2023-10-06T08:20:49Z" level=info msg=Trace args="[helm pull --destination /tmp/f81c643a-a279-4508-9d24-d95e49c58f6c --version 18.0.3 --repo https://charts.bitnami.com/bitnami harbor]" dir= operation_name="exec helm" time_ms=90000.921832
time="2023-10-06T08:20:49Z" level=error msg="finished unary call with code Unknown" error="`helm pull --destination /tmp/f81c643a-a279-4508-9d24-d95e49c58f6c --version 18.0.3 --repo https://charts.bitnami.com/bitnami harbor` failed timeout after 1m30s" grpc.code=Unknown grpc.method=GetAppDetails grpc.service=repository.RepoServerService grpc.start_time="2023-10-06T08:19:19Z" grpc.time_ms=90001.586 span.kind=server system=grpc

time="2023-10-06T08:26:12Z" level=info msg=Trace args="[helm pull --destination /tmp/1189b460-9cd8-46ef-b5f5-a32b50d2c07a --version 17.1.3 --repo https://charts.bitnami.com/bitnami harbor]" dir= operation_name="exec helm" time_ms=90001.43277
time="2023-10-06T08:26:12Z" level=error msg="finished unary call with code Unknown" error="`helm pull --destination /tmp/1189b460-9cd8-46ef-b5f5-a32b50d2c07a --version 17.1.3 --repo https://charts.bitnami.com/bitnami harbor` failed timeout after 1m30s" grpc.code=Unknown grpc.method=GetAppDetails grpc.service=repository.RepoServerService grpc.start_time="2023-10-06T08:24:42Z" grpc.time_ms=90002.09 span.kind=server system=grpc
``` 

This is on hetzner auction machine, but I doubt its something on network or OS since on the same node installing the same helm chart, or oven listing/updating works fine... its purely in ArgoCD where its not working. There is no reason to time out and  1m30s is a long time to get it, when helm on node takes 5s

**Comment by VladoPortos** [comment]
Issue is somehow related to Hetzner external firewall. I disabled the firewall ( Switched to accept all incoming connections ), and magically everything started to work. But this sux, what rule could possibly affect ArgoCD container getting helm stuff, while the very own node does not have issue... must be some back call or I have no idea...

**Reply by VladoPortos** [reply, marked as answer]
Needed to add `ack` for all ports 0-65535 for incoming and it seems to work now. ( added also for UDP just in case)

**Comment by alekc** [comment]
2 years later, this post is still valid with hetzner
